### PR TITLE
Prepare for Adding Segment Level UDQ Information to Restart File

### PR DESCRIPTION
**Pull request by bska**
In particular, identify more UDQ-related items in INTEHEAD and add more members to `InteHEAD::UdqParam` to hold counts of additional UDQ categories.

While here, also ensure that default-constructed objects of various `InteHEAD::*` structures have well-defined values.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

File: src/opm/output/eclipse/InteHEAD.cpp [hunk 1]
```diff
@@ -343,10 +343,10 @@ enum index : std::vector<int>::size_type {
   ih_285       =      285       ,              //       0
   MAX_ANALYTIC_AQUIFERS= VI::intehead::MAX_ANALYTIC_AQUIFERS, //  Declared maximum number of analytic aquifers in model.  AQUDIMS(5).
   ih_287       =      287       ,              //       0
-  ih_288       =      288       ,              //       0
-  ih_289       =      289       ,              //       0
-  NOIUADS       =      VI::intehead::NO_IUADS,  //       0
-  NOIUAPS       =      VI::intehead::NO_IUAPS,  //       0
+  NUM_AQU_UDQS =      VI::intehead::NO_AQU_UDQS,      //       0
+  NUM_BLK_UDQS =      VI::intehead::NO_AQU_UDQS,      //       0
```
Comment (blattms) [inline src/opm/output/eclipse/InteHEAD.cpp:347]: I am not into this code, but this strikes me. Is it on purpose that both entries get assigned `NO_AQU_UDQS`?

File: src/opm/output/eclipse/InteHEAD.cpp [hunk 2]
```diff
@@ -343,10 +343,10 @@ enum index : std::vector<int>::size_type {
...
+  NUM_IUADS    =      VI::intehead::NO_IUADS,  //       0
+  NUM_IUAPS    =      VI::intehead::NO_IUAPS,  //       0
```
Comment (blattms) [inline src/opm/output/eclipse/InteHEAD.cpp:349]: These two lines look indentical. Is that on purpose?

**Review by blattms — COMMENTED** [review]
I am not really into this code, but two things strike me

File: src/opm/output/eclipse/InteHEAD.cpp (hunk 1 above)
Comment (bska) [reply src/opm/output/eclipse/InteHEAD.cpp]: >  Is it on purpose that both entries get assigned `NO_AQU_UDQS`?

Good catch!  No, that's not on purpose&ndash;it's a mistake.  I've pushed an update to correct the mistake.

File: src/opm/output/eclipse/InteHEAD.cpp (hunk 2 above)
Comment (bska) [reply src/opm/output/eclipse/InteHEAD.cpp:349]: > These two lines look identical. Is that on purpose?

I don't quite see how these are identical.  Line 348 assigns `NUM_IUADS` (with a `D`) while line 349 assigns `NUM_IUAPS` (with a `P`).  Am I missing something?

**Comment by bska** [comment]
jenkins build this please

File: src/opm/output/eclipse/InteHEAD.cpp (hunk 2 above)
Comment (blattms) [reply src/opm/output/eclipse/InteHEAD.cpp:349]: Indeed. I should clean my glasses. :flushed: Was wondering a bit whether a compiler would even allow that.

**Comment by blattms** [comment]
Thanks a lot. Looks good now as far as I can tell (might be very short though)

**Pull request merged**